crates/sparrow-materialize/src/materialize.rs

```diff
@@ -0,0 +1,235 @@
+use error_stack::{IntoReport, ResultExt};
+use futures::{StreamExt, TryStreamExt};
+use sparrow_api::kaskada::v1alpha::{ComputePlan, ComputeTable, Destination, ProgressInformation};
+use sparrow_runtime::s3::S3Helper;
+use tokio::task::JoinHandle;
+
+#[derive(derive_more::Display, Debug, Clone, Copy)]
+pub enum Error {
+    #[display(fmt = "error during materialization process")]
+    Materialization,
+    #[display(fmt = "failed to create materialization")]
+    CreateMaterialization,
+    #[display(fmt = "failed to read progress")]
+    ReadingProgress,
+    #[display(fmt = "internal error")]
+    Internal,
+}
+
+impl error_stack::Context for Error {}
+
+pub struct Materialization {
```

Comment:
Rustdoc on the struct. What is it and how does it relate to materialization control, etc.